### PR TITLE
perf: Convert match breakpoints hook to provider to reduce calculation and event listeners

### PR DESCRIPTION
**Pull request by memoyil**

### CODE REVIEW

**Comment by chef-jojo** [comment]
can we move it back to uikit?

File: src/contexts/MatchBreakpoints/Provider.tsx [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+      return { ...accum, [size]: `(min-width: ${prevMinWidth}px)` }
+    }
+
+    const minWidth = prevMinWidth
+    const breakpoint = breakpointMap[size]
+
+    // Min width for next iteration
+    prevMinWidth = breakpoint + 1
+
+    return { ...accum, [size]: `(min-width: ${minWidth}px) and (max-width: ${breakpoint}px)` }
```
[the remark this answers is not in the document]
Comment (memoyil) [reply src/contexts/MatchBreakpoints/Provider.tsx:51]: I took the code from uikit, i think the bug exists in develop too should we fix it in here or another pr?

**Comment by memoyil** [comment]
> can we move it back to uikit?

Done

File: src/contexts/MatchBreakpoints/Provider.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (memoyil) [reply src/contexts/MatchBreakpoints/Provider.tsx]: I opened seperate pr https://github.com/pancakeswap/pancake-frontend/pull/4049

**Comment by memoyil** [comment]
> ah it's different to useMatchBreakpoints hooks. I think we can manage to merge them into one hook later

useMatchBreakpoints hooks used directly in uikit thats why we needed create another context hook. When uikit is merged into frontend, we can merge these hooks too

**Pull request merged**